src/utils/command.rs

```diff
@@ -100,13 +100,30 @@ pub use teloxide_macros::BotCommand;
 ///
 /// All variant attributes overlap the `enum` attributes.
 pub trait BotCommand: Sized {
-    fn try_from(s: &str) -> Option<Self>;
     fn descriptions() -> String;
-    fn parse<N>(s: &str, bot_name: N) -> Option<(Self, Vec<&str>)>
+    fn parse<N>(s: &str, bot_name: N) -> Result<Self, ParseError>
     where
         N: Into<String>;
 }
 
+#[derive(Debug)]
+pub enum ParseError {
+    LessArguments {
+        expected: u8,
+        found: u8,
+        message: String,
+    },
+    ManyArguments {
+        expected: u8,
+        found: u8,
+        message: String,
+    },
+    UncorrectFormat,
+    UncorrectCommand(String),
+    WrongBotName(String),
+    Custom(String)
```

Comment:
What does this variant (`Custom(String)`) mean? Could you please add documentation?